.github/workflows/callable_swap_tests.yml

```diff
@@ -0,0 +1,114 @@
+name: Callable Swap Functional tests
+
+on:
+  workflow_call:
+    inputs:
+      app_build_artifact:
+        required: true
+        type: string
+      exchange_build_artifact:
+        required: false
+        default: ''
+        description: 'If not provided, the workflow will build the Exchange app.'
+        type: string
+      ethereum_build_artifact:
+        required: false
+        default: ''
+        description: 'If not provided, the workflow will build the Ethereum app.'
+        type: string
+      regenerate_snapshots:
+        required: false
+        default: false
+        type: boolean
+
+env:
+  SWAP_TEST_DIR: "tests/swap"
+
+jobs:
+  build_exchange:
+    name: Build Exchange app using the reusable workflow
+    if: ${{ inputs.exchange_build_artifact == '' }}
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      app_repository: 'LedgerHQ/app-exchange'
+      app_branch_name: 'develop'
+      upload_app_binaries_artifact: "app_exchange_binaries"
+      use_case: 'use_test_keys'
+
+  build_ethereum:
+    name: Build Ethereum app using the reusable workflow
+    if: ${{ inputs.ethereum_build_artifact == '' }}
+    uses: LedgerHQ/ledger-app-workflows/.github/workflows/reusable_build.yml@v1
+    with:
+      app_repository: 'LedgerHQ/app-ethereum'
+      app_branch_name: 'develop'
+      upload_app_binaries_artifact: "app_ethereum_binaries"
+      upload_as_lib_artifact: "ethereum_"
+      use_case: 'use_test_keys'
+
+  get_artifacts_names:
+    name: Retrieve artifact names
+    needs: [build_exchange, build_ethereum]
+    if: ${{ always() }}
```

Comment:
why ?

Reply:
this job needs to be run every time, even if `build_exchange` and/or `build_ethereum` have been skipped...